packages/core/src/app.ts

```diff
@@ -119,11 +119,44 @@ export class Index extends Serializable {
         return this.items.filter((item) => item.hosts.some((h) => h === hashedHost)).length;
     }
 
+    splitHostBySubdomains(host: string) {
+        const parts = host.split(".");
+
+        // Ignore single domains
+        if (parts.length <= 2) {
+            return [];
+        }
+
+        // Remove the tld and domain from the parts, build it separately
+        const domain: string[] = [];
+
+        domain.unshift(parts.pop()!);
+        domain.unshift(parts.pop()!);
+
+        // Build list of subdomains to match, so given 'login.accounts.google.com', we'd see ['login', 'accounts'] as parts and ['google', 'com'] as domain, which should return ['google.com', 'accounts.google.com', and 'login.accounts.google.com']
+        return parts.reverse().reduce(
+            (currentDomainParts: string[], subdomain: string) => {
+                currentDomainParts.push(`${subdomain}.${currentDomainParts[currentDomainParts.length - 1]}`);
+
+                return currentDomainParts;
+            },
+            [domain.join(".")]
+        );
+    }
+
     async fuzzyMatchHost(host: string) {
+        const domains = this.splitHostBySubdomains(host);
+
+        const domainsMatches = (await Promise.all(domains.map(async (domain) => await this.matchHost(domain)))).reduce(
```

Comment:
Running this logic below would've been better (not checking if there was a direct host match), but a lot less readable. Your call.